tests/test_types.py

```diff
@@ -1806,6 +1806,13 @@ class Foo(BaseModel):
             a: type_ = Field('foo', title='A title', description='A description', **kwargs)
 
 
+@pytest.mark.xfail(
+    reason=(
+        "We allow python validation functions to wrap the type if the constraint isn't valid for the type. "
+        "An error isn't raised for an int or float annotated with this same invalid constraint, "
+        "so it's ok to mark this as xfail for now, but we should improve all of them in the future."
+    )
+)
```

Comment:
FWIW I do think we could special case something like `Annotated[float, Field(max_len=10)]` to raise an error at schema building time. If we're 100% sure the constraint makes no sense we can raise an error. But in the case of `Annoated[float, AfterValidator(...), Field(max_len=10)]` or whatever that we can't.

Reply:
Agreed, I think that should be addressed in a different PR though 👍